src/Entities/Player.cpp

```diff
@@ -89,6 +89,12 @@ cPlayer::cPlayer(cClientHandlePtr a_Client, const AString & a_PlayerName) :
 	m_UUID((a_Client != nullptr) ? a_Client->GetUUID() : ""),
 	m_CustomName("")
 {
+	if (a_PlayerName.empty() || a_PlayerName.length() > 16)
+	{
+		LOGERROR("Playername of %s is too long(>16) or empty", a_PlayerName.c_str());
+		VERIFY(false);
+	}
```

Comment:
Also, with this change, a malicious client could crash the server at any time by sending an empty username.